File: frontend/src/types/appointment.ts
export interface Appointment {
  id: number;
  user_id: number;
  office_id: number;
  service_id: number;
  counter_id: number | null;
  scheduled_time: string;
  status: 'pending' | 'confirmed' | 'checked_in' | 'in_progress' | 'completed' | 'cancelled' | 'no_show';
  qr_code_token: string;
  created_at: string;
  updated_at: string;
}

export interface AppointmentWithDetails extends Appointment {
  citizen_name: string;
  office_name: string;
  service_name: string;
  counter_name: string | null;
}

File: backend/app/schemas/appointment.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class AppointmentBase(BaseModel):
    office_id: int
    service_id: int
    scheduled_time: datetime

class AppointmentCreate(AppointmentBase):
    user_id: Optional[int] = None  # Optional, so staff/admin can book on behalf of a citizen

class AppointmentUpdate(BaseModel):
    status: Optional[str] = Field(None, pattern="^(pending|confirmed|checked_in|in_progress|completed|cancelled|no_show)$")
    counter_id: Optional[int] = None

class AppointmentResponse(AppointmentBase):
    id: int
    user_id: int
    counter_id: Optional[int]
    status: str
    qr_code_token: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class AppointmentWithDetailsResponse(AppointmentResponse):
    citizen_name: str
    office_name: str
    service_name: str
    counter_name: Optional[str] = None

    class Config:
        from_attributes = True
